### PR TITLE
(PDB-1139) Add preliminary 2.3.0 release notes

### PR DESCRIPTION
**Pull request by rbrw**
(WIP: not ready for merge.)

### CODE REVIEW

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -4,6 +4,151 @@ layout: default
...
+2.3.0
+-----
+
+PuppetDB 2.3.0 is a backwards-compatible release that adds support for
+Puppet 4.
+
+### Upgrading
+
+* For the best-possible performance and scaling capacity, we recommend
+  the latest version of PostgreSQL (9.3 or higher) with the
```
Comment (kbarber) [inline documentation/release_notes.markdown]: 9.4 is now the highest.

**Comment by rbrw** [comment]
@kbarber pushed a new version that hopefully addresses your concern with the postgresql version recommendations.

**Comment by wkalt** [comment]
+1

**Comment by wkalt** [comment]
this file is going to need review before release one way or another. Merging for now.

**Pull request merged**